device/api/umd/device/chip/local_chip.h

```diff
@@ -9,15 +9,20 @@
 #include "umd/device/chip/chip.h"
 
 namespace tt::umd {
+
+class TLBManager;
+
 class LocalChip : public Chip {
 public:
     LocalChip(tt_SocDescriptor soc_descriptor, int pci_device_id);
 
     TTDevice* get_tt_device() override;
+    TLBManager* get_tlb_manager() override;
 
     bool is_mmio_capable() const override;
 
 private:
     std::unique_ptr<TTDevice> tt_device_;
+    std::unique_ptr<TLBManager> tlb_manager_;
```

Comment:
I think I see TTDevice as something which is just there just so you don't have to read documentation to know how to use our card. And TTDevice should be something which is valid for tooling potentially as well, not just AI workload.

Also, please don't look at TLBManagement more of a Chip thing rather than a TTDevice thing. If the client of the TTDevice wants to have some kind of tlb helper, the client would create a TLBManager (which uses core TTDevice functions underneath) to program TLBs, and that exactly is Chip. Chip is a client of TTDevice. TLBManager should not be seen as something which is part of Chip.

I think you're suggesting that TLBManagement should have what it needs for programming TLBS, and then actually TTDevice would call TLB management, and not the other way around?

In any case, let's try to answer some questions to pinpoint the design. Let's pretend TLBManager doesn't exist, and we want to decide between TTDevice and Chip API:
- Which of these options is the desired API for TTDevice:
  - TTDevice offering write, accepting core. No tlb_index, no fallback_tlb. There is a flag in TTDevice constructor which sets up the default map of static tlbs. There is also a call to set_dynamic_tlb which allows user to setup custom TLB, and it will be saved to internal TTDevice map.
  - TTDevice offering write to tlb index. Doesn't care if you set up the tlb properly or not.
- In the regular simple user example, with the default TLB setup, which class has the bool flag "setup_default_tlbs" in its constructor, and then the constructor does something complex?
- Which class is the owner of core_to_tlb map?
- How does the scenario look like for a power user, which doesn't use Chip class? Power user meaning they want to program their own TLBs. Maybe they have some custom workload, and don't need static TLBs for cores how we handle them by default, maybe they setup most of TLBs as multicasts.

I'm obviously more aligned with Chip owning TLBmanager. I don't see at the moment any other way to separate cleanly into two structures, lower one which just provides syntactic sugar for configuring the device, and doesn't provide any security or robustness, and a higher one which should not allow you to break the device in any way.

Reply:
@pjanevskiTT and me discussed this offline.
I misunderstood the suggestion. The suggestion was only that TTDevice should own its TLBManager, rather than TLBManager having to be created on its own. I agree with this.
I misunderstood that TLBManager should be "underneath" TTDevice, so that TTDevice's API is in front of TLBManager.